depscan/lib/github.py

```diff
@@ -0,0 +1,106 @@
+from github import BadCredentialsException, Github, Auth
+from dataclasses import dataclass
+
+
+
+# Data class to represent a required GitHub personal access token scope for depscan
+@dataclass
+class RequiredTokenScope:
+    # Required token scope
+    scope: str
+    # If the required scope can be inherited from a parent scope, it is represented here
+    # Note that situation usually means that depscan will have the necessary permissions to function, but it will likely have more privilege than necessary
+    # For example, if a token has write:packages scope, then it inherits read:packages scope by default: 
+    #              > write:packages (Parent)
+    #              >>>> read:packages (Child)
+    scope_parent: str = None
+
+
+
+class GitHub:
+    # The GitHub personal access token required scopes for depscan
+    DEPSCAN_REQUIRED_TOKEN_SCOPES = [
+        RequiredTokenScope("read:packages", "write:packages")
```

Comment:
According to the [depscan documentation](https://github.com/owasp-dep-scan/dep-scan#github-security-advisory), `read:packages` is the one required scope on tokens. The scope of `write:packages` is a parent of read, such that if a token has write scope, it will also inherit read scope on packages.

Reply:
@timmyteo, could you kindly do some testing? The documentation might be outdated. Could you try using a token with no scope and see if the database is getting filled with advisories? We can definitely not request `write:packages` so that we operate with the least privilege.